add ban/unban user

diPhantxm opened the pull request:
- add ban/unban endpoints
- unit tests

nicklaw5 commented on moderation.go:
```diff
@@ -42,3 +42,68 @@ func (c *Client) GetBannedUsers(params *BannedUsersParams) (*BannedUsersResponse
+
+type BanUser struct {
+	BoardcasterId string `json:"broadcaster_id"`
+	CreatedAt     string `json:"created_at"`
+	EndTime       string `json:"end_time"`
```
Both `CreatedAt` and `EndTime` fields should be of type [Time](https://github.com/nicklaw5/helix/blob/main/time.go#L17-L19). This allows the our custom [UnmarshalJSON](https://github.com/nicklaw5/helix/blob/main/time.go#L26-L37) to handle converting the string represented time to a native [time.Time](https://pkg.go.dev/time#Time) type.

See the [User type](https://github.com/nicklaw5/helix/blob/main/users.go#L16) as an example.

nicklaw5 commented on moderation_test.go:
```diff
@@ -92,3 +93,187 @@ func TestGetBannedUsers(t *testing.T) {
+			}
+
+			if resp.Data.Bans[0].EndTime != "" {
+				layout := "2006-01-02T15:04:05Z"
+				createdTime, _ := time.Parse(layout, resp.Data.Bans[0].CreatedAt)
+				endTime, _ := time.Parse(layout, resp.Data.Bans[0].EndTime)
```
By actioning the above comment, you won't need to perform `time.Parse` here. The parsing occur when the JSON unmarshaling happens. These three lines can be removed.

nicklaw5 commented on moderation_test.go:
```diff
@@ -92,3 +93,187 @@ func TestGetBannedUsers(t *testing.T) {
+				createdTime, _ := time.Parse(layout, resp.Data.Bans[0].CreatedAt)
+				endTime, _ := time.Parse(layout, resp.Data.Bans[0].EndTime)
+
+				expireTime := createdTime.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))
+
+				if !expireTime.Equal(endTime) {
+					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
+				}
```
These lines can then be written like so:

```suggestion
				expireTime := resp.Data.Bans[0].CreatedAt.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))

				if !expireTime.Equal(resp.Data.Bans[0].EndTime) {
					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
				}
```

nicklaw5 reviewed (changes requested):
Thanks for contributing. Looks good. We just need to adjust the time field types. See the comments I've left.

diPhantxm commented:
Thanks for your comment. I've fixed data types as you said. Also added comment to unban user function and noticed one more defect in test function.
Didn't notice your data type for time right away. It's actually my first time contributing to an open source project.

nicklaw5 commented:
Thanks and welcome to open source development :slightly_smiling_face:
